actor-tests/src/test/scala/org/apache/pekko/util/ByteStringSpec.scala

```diff
@@ -814,6 +814,13 @@ class ByteStringSpec extends AnyWordSpec with Matchers with Checkers {
         }
       }
 
+      "created from iterator" in {
```

Comment:
This test is here to make sure that the `apply` method for Iterator works for both Scala 2.12 and Scala 2.13. Without the added `Iterator[Byte]` apply method in this PR, this test will fail to compile.